silence `lll` linter for imports

### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

Currently the `lll` linter warns for long import lines.

This behaviour should be discontinued, since `import` paths cannot really be influenced much by changing code layout.
The maximum optimization to be done is to reduce or drop import aliases.


### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.49.0 built from (unknown, mod sum: "h1:I8WHOavragDttlLHtSraHn/h39C+R60bEQ5NoGcHQr8=") on (unknown)
```

</details>

### Configuration file

Default config with `lll` config as follows:

```yaml
  lll:
    # max line length, lines longer will be reported. Default is 120.
    # '\t' is counted as 1 character by default, and can be changed with the tab-width option
    line-length: 120
    # tab width in spaces. Default to 1.
    tab-width: 1
```

### Go environment

<details>

```console
$ go version && go env
go version go1.19.1 darwin/amd64
```

</details>

### Verbose output of running

<details>

```console
go run github.com/golangci/golangci-lint/cmd/golangci-lint cache clean
# github.com/golangci/golangci-lint/cmd/golangci-lint
ld: warning: -no_pie is deprecated when targeting new OS versions
go run github.com/golangci/golangci-lint/cmd/golangci-lint run -v
# github.com/golangci/golangci-lint/cmd/golangci-lint
ld: warning: -no_pie is deprecated when targeting new OS versions
INFO [config_reader] Config search paths: [./ /some-path-on-my-system /some-other-path-on-my-system /some-path-on-my-system /] 
INFO [config_reader] Used config file .golangci.yaml 
INFO [lintersdb] Active 8 linters: [errcheck gosimple govet ineffassign lll staticcheck typecheck unused] 
INFO [loader] Go packages loading at mode 575 (compiled_files|deps|files|types_sizes|exports_file|imports|name) took 840.030995ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 1.516694ms 
INFO [linters context/goanalysis] analyzers took 49.397352601s with top 10 stages: buildir: 37.133830681s, inspect: 2.095002854s, fact_deprecated: 1.95023797s, printf: 1.63728323s, ctrlflow: 1.629920841s, fact_purity: 1.483155536s, nilness: 1.23624039s, SA5012: 1.127976062s, typedness: 1.064085254s, unsafeptr: 12.579108ms 
INFO [runner] Processors filtering stat (out/in): nolint: 1/1, max_same_issues: 1/1, skip_files: 1/1, skip_dirs: 1/1, exclude-rules: 1/1, max_from_linter: 1/1, severity-rules: 1/1, filename_unadjuster: 1/1, path_prettifier: 1/1, autogenerated_exclude: 1/1, max_per_file_from_linter: 1/1, path_prefixer: 1/1, sort_results: 1/1, uniq_by_line: 1/1, diff: 1/1, source_code: 1/1, path_shortener: 1/1, cgo: 1/1, identifier_marker: 1/1, exclude: 1/1 
INFO [runner] processing took 541.933µs with stages: nolint: 314.274µs, path_prettifier: 58.8µs, autogenerated_exclude: 45.776µs, source_code: 44.537µs, identifier_marker: 28.568µs, skip_dirs: 25.858µs, exclude-rules: 12.902µs, uniq_by_line: 2.462µs, max_same_issues: 1.952µs, cgo: 1.407µs, path_shortener: 1.046µs, max_from_linter: 899ns, filename_unadjuster: 847ns, exclude: 631ns, skip_files: 541ns, max_per_file_from_linter: 513ns, diff: 251ns, sort_results: 241ns, severity-rules: 240ns, path_prefixer: 188ns 
INFO [runner] linters took 16.801183214s with stages: goanalysis_metalinter: 16.80054929s 
lambda/cmd/lambda.go:19: line is 128 characters (lll)
       "gitlab.my-long-hostname.com/my-long-organization/my-long-project-group/my-long-project-name/lambda/internal/pkg/eventhandler"
INFO File cache stats: 1 entries of total size 4.0KiB 
INFO Memory: 163 samples, avg is 469.1MB, max is 1194.3MB 
INFO Execution took 17.659406238s                 
exit status 1
make: *** [lint] Error 1
```

</details>

### Code example or link to a public repository

<details>

```console
```

</details>

--- a/pkg/golinters/lll.go
+++ b/pkg/golinters/lll.go
@@ -84,17 +84,18 @@ func getLLLIssuesForFile(filename string, maxLineLen int, tabSpaces string) ([]r
 	}
 	defer f.Close()
 
-	lineNumber := 1
+	lineNumber := 0
 	multiImportEnabled := false
 
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
+		lineNumber++
+
 		line := scanner.Text()
 		line = strings.ReplaceAll(line, "\t", tabSpaces)
 
 		if strings.HasPrefix(line, "import") {
 			multiImportEnabled = strings.HasSuffix(line, "(")
-			lineNumber++
 			continue
 		}
 
@@ -103,7 +104,6 @@ func getLLLIssuesForFile(filename string, maxLineLen int, tabSpaces string) ([]r
 				multiImportEnabled = false
 			}
 
-			lineNumber++
 			continue
 		}
 
@@ -118,7 +118,6 @@ func getLLLIssuesForFile(filename string, maxLineLen int, tabSpaces string) ([]r
 				FromLinter: lllName,
 			})
 		}
-		lineNumber++
 	}
 
 	if err := scanner.Err(); err != nil {

--- a/pkg/golinters/lll.go
+++ b/pkg/golinters/lll.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"go/token"
 	"os"
+	"regexp"
 	"strings"
 	"sync"
 	"unicode/utf8"
@@ -18,6 +19,9 @@ import (
 )
 
 const lllName = "lll"
+
+var lllMultiImportStartRegexp = regexp.MustCompile(`^import \($`)
+var lllSingleImportRegexp = regexp.MustCompile(`^import \".+\"$`)
 
 //nolint:dupl
 func NewLLL(settings *config.LllSettings) *goanalysis.Linter {
@@ -86,9 +90,33 @@ func getLLLIssuesForFile(filename string, maxLineLen int, tabSpaces string) ([]r
 
 	lineNumber := 1
 	scanner := bufio.NewScanner(f)
+	importsEnded := false
+	multiImportEnabled := false
 	for scanner.Scan() {
 		line := scanner.Text()
 		line = strings.ReplaceAll(line, "\t", tabSpaces)
+		// Skips imports
+		if !importsEnded {
+			if lllSingleImportRegexp.MatchString(line) {
+				lineNumber++
+				continue
+			}
+
+			if lllMultiImportStartRegexp.MatchString(line) {
+				multiImportEnabled = true
+				lineNumber++
+				continue
+			}
+
+			if multiImportEnabled && line == ")" {
+				importsEnded = true
+				lineNumber++
+				continue
+			}
+
+			importsEnded = true
+		}
+
 		lineLen := utf8.RuneCountInString(line)
 		if lineLen > maxLineLen {
 			res = append(res, result.Issue{

--- a/pkg/golinters/lll.go
+++ b/pkg/golinters/lll.go
@@ -85,19 +85,19 @@ func getLLLIssuesForFile(filename string, maxLineLen int, tabSpaces string) ([]r
 	defer f.Close()
 
 	lineNumber := 1
-	scanner := bufio.NewScanner(f)
 	multiImportEnabled := false
+
+	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
 		line := scanner.Text()
 		line = strings.ReplaceAll(line, "\t", tabSpaces)
-		if strings.HasPrefix(line, "import") {
-			if strings.HasSuffix(line, "(") {
-				multiImportEnabled = true
-			}
 
+		if strings.HasPrefix(line, "import") {
+			multiImportEnabled = strings.HasSuffix(line, "(")
 			lineNumber++
 			continue
 		}
+
 		if multiImportEnabled {
 			if line == ")" {
 				multiImportEnabled = false

--- a/pkg/golinters/lll.go
+++ b/pkg/golinters/lll.go
@@ -86,28 +86,25 @@ func getLLLIssuesForFile(filename string, maxLineLen int, tabSpaces string) ([]r
 
 	lineNumber := 1
 	scanner := bufio.NewScanner(f)
-	importsEnded := false
 	multiImportEnabled := false
 	for scanner.Scan() {
 		line := scanner.Text()
 		line = strings.ReplaceAll(line, "\t", tabSpaces)
-		// Skips imports
-		if !importsEnded {
-			if strings.HasPrefix(line, "import") {
-				if strings.HasSuffix(line, "(") {
-					multiImportEnabled = true
-				}
-
-				lineNumber++
-				continue
+		if strings.HasPrefix(line, "import") {
+			if strings.HasSuffix(line, "(") {
+				multiImportEnabled = true
 			}
 
-			if multiImportEnabled && line != ")" {
-				lineNumber++
-				continue
+			lineNumber++
+			continue
+		}
+		if multiImportEnabled {
+			if line == ")" {
+				multiImportEnabled = false
 			}
 
-			importsEnded = true
+			lineNumber++
+			continue
 		}
 
 		lineLen := utf8.RuneCountInString(line)

--- a/pkg/golinters/lll.go
+++ b/pkg/golinters/lll.go
@@ -5,7 +5,6 @@ import (
 	"fmt"
 	"go/token"
 	"os"
-	"regexp"
 	"strings"
 	"sync"
 	"unicode/utf8"
@@ -19,9 +18,6 @@ import (
 )
 
 const lllName = "lll"
-
-var lllMultiImportStartRegexp = regexp.MustCompile(`^import \($`)
-var lllSingleImportRegexp = regexp.MustCompile(`^import \".+\"$`)
 
 //nolint:dupl
 func NewLLL(settings *config.LllSettings) *goanalysis.Linter {
@@ -97,19 +93,16 @@ func getLLLIssuesForFile(filename string, maxLineLen int, tabSpaces string) ([]r
 		line = strings.ReplaceAll(line, "\t", tabSpaces)
 		// Skips imports
 		if !importsEnded {
-			if lllSingleImportRegexp.MatchString(line) {
+			if strings.HasPrefix(line, "import") {
+				if strings.HasSuffix(line, "(") {
+					multiImportEnabled = true
+				}
+
 				lineNumber++
 				continue
 			}
 
-			if lllMultiImportStartRegexp.MatchString(line) {
-				multiImportEnabled = true
-				lineNumber++
-				continue
-			}
-
-			if multiImportEnabled && line == ")" {
-				importsEnded = true
+			if multiImportEnabled && line != ")" {
 				lineNumber++
 				continue
 			}

--- a/test/testdata/configs/lll_import.yml
+++ b/test/testdata/configs/lll_import.yml
@@ -1,0 +1,4 @@
+linters-settings:
+  lll:
+    tab-width: 4
+    line-length: 60

--- a/test/testdata/lll_multi_import.go
+++ b/test/testdata/lll_multi_import.go
@@ -1,0 +1,14 @@
+//golangcitest:args -Elll
+//golangcitest:config_path testdata/configs/lll_import.yml
+//golangcitest:expected_exitcode 0
+package testdata
+
+import (
+	anotherVeryLongImportAliasNameForTest "github.com/golangci/golangci-lint/internal/golinters"
+	veryLongImportAliasNameForTest "github.com/golangci/golangci-lint/internal/golinters"
+)
+
+func LllMultiImport() {
+	_ = veryLongImportAliasNameForTest.NewLLL(nil)
+	_ = anotherVeryLongImportAliasNameForTest.NewLLL(nil)
+}

--- a/test/testdata/lll_single_import.go
+++ b/test/testdata/lll_single_import.go
@@ -1,0 +1,10 @@
+//golangcitest:args -Elll
+//golangcitest:config_path testdata/configs/lll_import.yml
+//golangcitest:expected_exitcode 0
+package testdata
+
+import veryLongImportAliasNameForTest "github.com/golangci/golangci-lint/internal/golinters"
+
+func LllSingleImport() {
+	_ = veryLongImportAliasNameForTest.NewLLL(nil)
+}

